backend/src/routes/helpers/validation.ts

```diff
@@ -0,0 +1,9 @@
+import * as z from 'zod';
```

Comment:
This checks that the incoming input is a non empty string, runs the transform function, then check that the transformed output is a number. Final value is of number type, or error is thrown.